slobrok/src/vespa/slobrok/server/cmd.cpp

```diff
@@ -65,63 +71,60 @@ void
 RegRpcSrvCommand::cleanupReservation()
 {
     RpcServerMap &map = _data->env._rpcsrvmap;
-    ReservedName *rsvp = map.getReservation(_data->name.c_str());
-    if (rsvp != NULL && rsvp->isLocal) {
+    const ReservedName *rsvp = map.getReservation(_data->name.c_str());
+    if (rsvp != nullptr && rsvp->isLocal) {
         map.removeReservation(_data->name.c_str());
     }
 }
 
 void
 RegRpcSrvCommand::doneHandler(OkState result)
 {
-    LOG_ASSERT(_data != NULL);
-
+    LOG_ASSERT(_data != nullptr);
+    RegRpcSrvData & data = *_data;
     if (result.failed()) {
-        LOG(warning, "failed in state %d: %s",
-            _data->_state, result.errorMsg.c_str());
+        LOG(warning, "failed in state %d: %s", data._state, result.errorMsg.c_str());
         cleanupReservation();
         // XXX should handle different state errors differently?
-        if (_data->registerRequest != NULL) {
-            _data->registerRequest->SetError(FRTE_RPC_METHOD_FAILED,
-                                             result.errorMsg.c_str());
-            _data->registerRequest->Return();
+        if (data.registerRequest != nullptr) {
+            data.registerRequest->SetError(FRTE_RPC_METHOD_FAILED, result.errorMsg.c_str());
+            data.registerRequest->Return();
         } else {
             LOG(warning, "ignored: %s", result.errorMsg.c_str());
         }
         goto alldone;
     }
-    if (_data->_state == RegRpcSrvData::RDC_INIT) {
+    if (data._state == RegRpcSrvData::RDC_INIT) {
         LOG(spam, "phase wantAdd(%s,%s)",
-            _data->name.c_str(), _data->spec.c_str());
-        _data->_state = RegRpcSrvData::XCH_WANTADD;
-        _data->env._exchanger.wantAdd(_data->name.c_str(), _data->spec.c_str(), *this);
+            data.name.c_str(), data.spec.c_str());
+        data._state = RegRpcSrvData::XCH_WANTADD;
+        data.env._exchanger.wantAdd(data.name.c_str(), data.spec.c_str(), std::move(*this));
         return;
-    } else if (_data->_state == RegRpcSrvData::XCH_WANTADD) {
+    } else if (data._state == RegRpcSrvData::XCH_WANTADD) {
         LOG(spam, "phase addManaged(%s,%s)",
-            _data->name.c_str(), _data->spec.c_str());
-        _data->_state = RegRpcSrvData::CHK_RPCSRV;
-        _data->env._rpcsrvmanager.addManaged(_data->name.c_str(), _data->spec.c_str(), *this);
+            data.name.c_str(), data.spec.c_str());
+        data._state = RegRpcSrvData::CHK_RPCSRV;
+        data.env._rpcsrvmanager.addManaged(data.name, data.spec.c_str(), std::move(*this));
```

Comment:
move data (unique_ptr on stack) instead of self

Reply:
Fixed